### DISCUSSION TITLE
Should Retryer catch Exception instead of Throwable?

### DISCUSSION THREAD

**Discussion by sleberknight** [kiwiproject/retrying-again#35] [General]
Sonar is very unhappy that `Retryer#call` catches `Throwable` per rule [java:S1181](https://rules.sonarsource.com/java/RSPEC-1181) (Throwable and Error should not be caught)

> Throwable is the superclass of all errors and exceptions in Java. Error is the superclass of all errors, which are not meant to be caught by applications.
>
> Catching either Throwable or Error will also catch OutOfMemoryError and InternalError, from which an application should not attempt to recover.

So, should we change `Retryer` so it only catches `Exception`?

**Comment by chrisrohr** [comment]
Yes, go ahead, we won't be able to much with things like OutOfMemoryError etc

**Reply by sleberknight** [reply]
Created issue #54 . This discussion is resolved.
